File: sumOfDigits.c
#include <stdio.h>

int main()
{
    int number,lastdigit,copyNum,sum=0;
    printf("Enter a number: ");
    scanf("%d",&number);
    copyNum=number;

    while (number!=0)
    {
        lastdigit=number%10;
        sum = sum+lastdigit;
        number/=10;
    }
   printf("The sum of digits in %d is %d\n",copyNum,sum);
    
     
    return 0;
}

File: Sumofdigits.java
// Sum of digits till it results in single digit.

import java.util.Scanner;

public class Sumofdigits {

	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);

		System.out.println("Enter a number : ");

		int input = sc.nextInt();
		int result = additionOfDigits(input);
		if(result>9)
			result = additionOfDigits(result);
		System.out.println(result);

		
	}
	private static int additionOfDigits(int input)
	{
	int total=0;
  
		while(input>0) {

			total = total + input%10;
			input/=10;

		}

		return total;
	}

}